### PR TITLE
Native slash commands that replace old commands

### PR DESCRIPTION
**Pull request by Saagar3690**
Replaced all old commands with native slash commands.

TODO:
- [x] FIx ephemeral for whoami and lookup
- [x] Fix DiscordAPIError for some commands
- [x] Restrict slash commands to bot-commands, moderators, and DMs
- [x] More Testing!!!

### CODE REVIEW

**Comment by Saagar3690** [comment]
Been a while, but finally finished up slash commands. All commands (iam, verify, name, year, major, transfer, whoami) should work as well as some extra commands for admins and mods. Some important things to note:

1. Commands should be only visible to the person who uses it (please let me know if you ever run into a situation where this is not the case)
2. Slash commands are restricted to only bot commands and moderators. If used in any other channel, it will tell the user to retry the command in one of those channels.
3. There will be a DiscordAPIError if the bot is used by an admin/mod to try and change the nickname of someone with a higher role than bot (but I have accounted for this using try catch and an error message), but otherwise it should work (I haven't been able to do this since my role in the test server is above the bot's role).

I have tested as much as possible but per always, there's room to test some more in case I missed some edge cases or anything else (which I most likely did). Please let me know any bugs or issues you face, so I can fix it right away.

**Comment by entrohpy** [comment]
I'll try to take a look at this by next week (July 19). Thanks for the fantastic work!

File: main.js
Comment (advaithg) [inline main.js:415]: I don't think there should be an `isModOrAdmin(member)` here, since that makes help not work unless you're a mod or admin!

**Review by advaithg — APPROVED** [review]
Looks good to me! 
Probably a good idea to wait on @entrohpy's review as well before merging

File: .eslintrc.json
Comment (entrohpy) [inline .eslintrc.json:18]: just a personal preference. indents case statements in switch blocks

File: commands/getUser.js
Comment (entrohpy) [inline commands/getUser.js:63]: The new line between the backticks and text actually shows because backticks are for string "literals".

Comment (entrohpy) [reply commands/getUser.js:63]: this is an issue that exists in master as well. Took the opportunity to fix it rn

File: commands/setUser.js
Comment (entrohpy) [inline commands/setUser.js:347]: can we hide commands from users that don't have the specific role (moderator)?

File: main.js
Comment (entrohpy) [inline main.js:114]: added finite choices instead of free response

**Comment by entrohpy** [comment]
@Saagar3690 @advaithg given my extremely long turnaround time on this PR, I decided to make edits myself. I explain them as comments above. Please triage them and either react/reply to this comment once you're done.

**Comment by entrohpy** [comment]
Awesome, I'll take another look this week and merge it into a feature branch. Once we're ready to deploy, we can switch branches on the prod server. After an initial test run, we can merge it into master

**Comment by entrohpy** [comment]
amazing work team

**Pull request merged**
